### PR TITLE
IDENTITY-5502 : Make NameIDFormat configurable in SAML SSO federated authenticator

### PR DESCRIPTION
**Pull request by mefarazath**
This is also related to https://wso2.org/jira/browse/IDENTITY-5597

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/util/IdentityApplicationConstants.java [hunk 1]
```diff
@@ -130,6 +130,9 @@ private OpenID() {
             public static final String ATTRIBUTE_CONSUMING_SERVICE_INDEX = "AttributeConsumingServiceIndex";
             public static final String DESTINATION_URL_PREFIX = "DestinationURI";
 
+            public static final String UNSPECIFIED_NAME_ID_FORMAT =
+                    "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified";
```
Comment (omindu) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/util/IdentityApplicationConstants.java]: Can't we use the constants exposed in OpenSAML here ?

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp [hunk 2]
```diff
@@ -3667,6 +3694,18 @@
                                         </div>
                                     </td>
                                 </tr>
+                                <tr>
+                                    <td class="leftCol-med labelField"><fmt:message key='name.id.format'/>:<span
+                                            class="required">*</span></td>
+                                    <td>
+                                        <input id="NameIDType" name="NameIDType" type="text" size="70"
```
Comment (omindu) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: I'm assuming this is the place where we enter NameIDType in UI. Any idea whether the number of NameIDTypes  is fixed or not ? If it's fixed we can use a combo box.

Comment (mefarazath) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: There are 8 standard ones. But the spec doesn't limit anyone from using custom NameId Formats.
That was the reasoning behind making this a text box.

Comment (omindu) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: Seems this is a mandatory field. An editable combobox with the defaults would be ideal ;).

**Pull request merged**